lib/ReactGridLayout.jsx

```diff
@@ -429,31 +461,17 @@ export default class ReactGridLayout extends React.Component<Props, State> {
           if (Number.isFinite(leastY)) l.h = leastY - l.y;
         }
       }
+      l.w = w;
+      l.h = h;
 
       return l;
     });
 
     // Shouldn't ever happen, but typechecking makes it necessary
     if (!l) return;
 
-    let finalLayout;
```

Comment:
I moved the majority of this logic to within `withLayoutItem`. This probably should have been within there to begin with, so sorry about missing that originally.